### ISSUE TITLE
[proposal] optional `go mod tidy` run on `go generate`

### ISSUE THREAD

**Issue by yar00001** [99designs/gqlgen#1643]
### What happened?
[opened this issue based on Contribution Guidelines](https://github.com/99designs/gqlgen/blob/master/CONTRIBUTING.md)

`go mod tidy` is run by default [in the latest gqlgen release](https://github.com/99designs/gqlgen/pull/1501/files). This might [cause failures](https://github.com/99designs/gqlgen/issues/1642) in repos containing multiple projects that we don't have access to.

### What did you expect?

`go mod tidy` can run optionally, based on user's configuration.

**Comment by ddevault** [comment]
This breaks my repo by causing go generate ./... to fail every subsequent time unless I go get all of the gqlgen stuff again.

**Comment by ddevault** [comment]
More generally speaking, what place does gqlgen have to tidy my go.mod anyway? It's my go.mod, not yours. I'll run go mod tidy myself if I want to.

**Comment by titus-zx** [comment]
any update on this?

**Comment by suhabe** [comment]
also facing this issue

**Comment by abhinavnair** [comment]
https://github.com/99designs/gqlgen/pull/1644 solves the issue. And the good news is that it has been finally released in [v0.15.0 ](https://github.com/99designs/gqlgen/releases/tag/v0.15.0) 🎉 
Thanks @yar00001 @StevenACoffman 🙇🏻‍♂️

**Comment by adam-hanna** [comment]
Still seeing this issue in v0.17.20

```
$ gqlgen version  
v0.17.20
➜  $ go mod tidy -compat=1.17
➜  $ go run github.com/99designs/gqlgen generate
go: downloading github.com/google/go-cmp v0.5.9
go: downloading github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1
github.com/adam-hanna/bitmex-market-maker/pkg/broker/dydx imports
        google.golang.org/grpc tested by
        google.golang.org/grpc.test imports
        github.com/google/go-cmp/cmp loaded from github.com/google/go-cmp@v0.5.8,
        but go 1.16 would select v0.5.9
github.com/adam-hanna/bitmex-market-maker/cmd/eth/commands imports
        github.com/ethereum/go-ethereum/crypto imports
        github.com/btcsuite/btcd/btcec/v2/ecdsa tested by
        github.com/btcsuite/btcd/btcec/v2/ecdsa.test imports
        github.com/btcsuite/btcd/chaincfg/chainhash loaded from github.com/btcsuite/btcd@v0.20.1-beta,
        but go 1.16 would fail to locate it:
        ambiguous import: found package github.com/btcsuite/btcd/chaincfg/chainhash in multiple modules:
        github.com/btcsuite/btcd v0.20.1-beta (/home/adamhanna/.gvm/pkgsets/go1.18.7/global/pkg/mod/github.com/btcsuite/btcd@v0.20.1-beta/chaincfg/chainhash)
        github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1 (/home/adamhanna/.gvm/pkgsets/go1.18.7/global/pkg/mod/github.com/btcsuite/btcd/chaincfg/chainhash@v1.0.1)
github.com/adam-hanna/bitmex-market-maker/pkg/broker/dydx imports
        google.golang.org/grpc imports
        google.golang.org/grpc/internal/balancer/gracefulswitch tested by
        google.golang.org/grpc/internal/balancer/gracefulswitch.test imports
        github.com/google/go-cmp/cmp/cmpopts loaded from github.com/google/go-cmp@v0.5.8,
        but go 1.16 would select v0.5.9
github.com/adam-hanna/bitmex-market-maker/pkg/broker/dydx imports
        google.golang.org/grpc tested by
        google.golang.org/grpc.test imports
        github.com/google/go-cmp/cmp imports
        github.com/google/go-cmp/cmp/internal/diff loaded from github.com/google/go-cmp@v0.5.8,
        but go 1.16 would select v0.5.9
github.com/adam-hanna/bitmex-market-maker/pkg/broker/dydx imports
        google.golang.org/grpc tested by
        google.golang.org/grpc.test imports
        github.com/google/go-cmp/cmp imports
        github.com/google/go-cmp/cmp/internal/flags loaded from github.com/google/go-cmp@v0.5.8,
        but go 1.16 would select v0.5.9
github.com/adam-hanna/bitmex-market-maker/pkg/broker/dydx imports
        google.golang.org/grpc tested by
        google.golang.org/grpc.test imports
        github.com/google/go-cmp/cmp imports
        github.com/google/go-cmp/cmp/internal/function loaded from github.com/google/go-cmp@v0.5.8,
        but go 1.16 would select v0.5.9
github.com/adam-hanna/bitmex-market-maker/pkg/broker/dydx imports
        google.golang.org/grpc tested by
        google.golang.org/grpc.test imports
        github.com/google/go-cmp/cmp imports
        github.com/google/go-cmp/cmp/internal/value loaded from github.com/google/go-cmp@v0.5.8,
        but go 1.16 would select v0.5.9

To upgrade to the versions selected by go 1.16, leaving some packages unresolved:
        go mod tidy -e -go=1.16 && go mod tidy -e -go=1.17
If reproducibility with go 1.16 is not needed:
        go mod tidy -compat=1.17
For other options, see:
        https://golang.org/doc/modules/pruning
tidy failed: go mod tidy failed: exit status 1
exit status 1
```
